pkg/cloud/aws/sns/topic.go

```diff
@@ -80,6 +85,82 @@ func (t *snsTopic) Publish(ctx context.Context, msg string, attributes ...map[st
 	return err
 }
 
+// PublishBatch fails at the first batch that could not be published.
+func (t *snsTopic) PublishBatch(ctx context.Context, messages []string, attributes []map[string]interface{}) error {
+	if len(messages) != len(attributes) {
+		return fmt.Errorf("there should be as many attributes as messages")
+	}
+
+	entries, err := t.computeEntries(messages, attributes)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(messages); i += MaxBatchSize {
+		currentBatch := t.getSubSlice(i, entries)
+
+		err := t.publishSubSlice(ctx, currentBatch)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *snsTopic) getSubSlice(from int, entries []types.PublishBatchRequestEntry) []types.PublishBatchRequestEntry {
+	endIndex := from + MaxBatchSize
+	if endIndex >= len(entries) {
+		endIndex = len(entries)
+	}
+
+	return entries[from:endIndex]
+}
+
+func (t *snsTopic) publishSubSlice(ctx context.Context, entries []types.PublishBatchRequestEntry) error {
+	if len(entries) > MaxBatchSize {
+		return fmt.Errorf("batch is too large, max length is %d", MaxBatchSize)
+	}
+
+	input := &sns.PublishBatchInput{
+		TopicArn:                   aws.String(t.topicArn),
+		PublishBatchRequestEntries: entries,
+	}
+
+	_, err := t.client.PublishBatch(ctx, input)
+
+	if exec.IsRequestCanceled(err) {
+		t.logger.WithContext(ctx).WithFields(log.Fields{
+			"arn": t.topicArn,
+		}).Info("request was canceled while publishing to topic")
+
+		return err
+	}
+
+	return err
+}
+
+func (t *snsTopic) computeEntries(messages []string, attributes []map[string]interface{}) ([]types.PublishBatchRequestEntry, error) {
+	result := make([]types.PublishBatchRequestEntry, len(messages))
+
+	for i := 0; i < len(messages); i++ {
+		messageAttributes, err := buildAttributes([]map[string]interface{}{attributes[i]})
+		if err != nil {
+			t.logger.Error("can not build message attributes %w", err)
```

Comment:
we shouldn't log the error here..  
logging errors should always be done by the caller of PublishBatch